MoreLinq/Memoize.cs

```diff
@@ -30,6 +30,10 @@ static partial class MoreEnumerable
         /// </summary>
         /// <param name="source">The source sequence.</param>
         /// <returns>Returns a sequence that corresponds to a cached version of the input sequence.</returns>
+        /// <remarks>
+        /// Enumeration of the result sequence is thread-safe, that is, multiple independent IEnumerator values 
```

Comment:
> I would also remark that the source should be thread-agnostic as MoveNext calls into it can potentially come from different threads otherwise the behavior is undefined.

@atifaziz  sorry, I dont understand what you mean...  can you explain me again? maybe use other words for better undestanding... 

also could you help me with the remarks? i am not so good with documentation.